src/main/kotlin/org/jetBrains/research/tasktracker/tracking/DefaultContentProvider.kt

```diff
@@ -54,13 +54,25 @@ object DefaultContentProvider {
         else -> ""
     }
 
-    private fun getPackage(extension: Extension): String {
-        val currentPackage =
-            "package $PLUGIN_NAME.${extension.name.lowercase(Locale.getDefault())}"
-        return when (extension) {
-            Extension.JAVA -> "$currentPackage;$lineSeparator$lineSeparator"
-            Extension.KOTLIN -> "$currentPackage$lineSeparator$lineSeparator"
+    private fun getPackage(task: Task): String {
+        val pack = "package $PLUGIN_NAME.${relativePathToPackage(task) ?: lowercaseExtension(task)}"
+        return when (task.getExtension()) {
+            Extension.JAVA -> "$pack;$lineSeparator$lineSeparator"
+            Extension.KOTLIN -> "$pack$lineSeparator$lineSeparator"
             else -> ""
         }
     }
+
+    private fun relativePathToPackage(task: Task): String? {
+        var pack = task.getRelativeFilePath()?.replace("/", ".") ?: return null
+        if (pack.first() == '/') {
+            pack = pack.substring(1)
+        }
+        if (pack.last() == '/') {
+            pack = pack.dropLast(1)
+        }
```

Comment:
Let's use currentPackage" or just "package" here

Reply:
Also, if you replaced "/" into ".", how the pack can start from  "/"...? you probably need to rewrite it with something like this:
```kotlin
    private fun relativePathToPackage(task: Task) =
        task.getRelativeFilePath()
            ?.removePrefix("/")
            ?.removeSuffix("/")
            ?.replace("/", ".")
```